src/coreclr/jit/lsraarm64.cpp

```diff
@@ -1011,40 +1011,49 @@ int LinearScan::BuildHWIntrinsic(GenTreeHWIntrinsic* intrinsicTree)
 
     if (intrin.op1 != nullptr)
     {
-        bool simdRegToSimdRegMove = false;
-
-        if ((intrin.id == NI_Vector64_CreateScalarUnsafe) || (intrin.id == NI_Vector128_CreateScalarUnsafe))
-        {
-            simdRegToSimdRegMove = varTypeIsFloating(intrin.op1);
-        }
-        else if (intrin.id == NI_AdvSimd_Arm64_DuplicateToVector64)
+        // Do not give a register to the second operand if it is contained.
+        if (!hasImmediateOperand && intrin.op1->isContained())
```

Comment:
I think the comment here should say first operand?

It will end up becoming the second operand, since its commutative; but there hasn't been any kind of swapping yet to put it in the right place (looks like that is happening in codegen).

Reply:
Yea, that's a typo - it should say 'first operand'.

As far as swapping, this PR does handle it: https://github.com/dotnet/runtime/pull/62933/files#diff-45c9473f1b1d031652a13186a211c1aef1406a74f576d5f04e3a65766b0f5050R500